Main/src/main/java/com/main/acad/util/ConnectionPool.java

```diff
@@ -1,37 +1,50 @@
 package com.main.acad.util;
 
-import org.apache.commons.dbcp.BasicDataSource;
-
-import javax.sql.DataSource;
-import java.io.IOException;
-import java.io.InputStream;
-import java.util.Properties;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
 
 public class ConnectionPool {
+    private int maxPoolSize = 10;
+    private int initialPoolSize = 1;
+    private int currentPoolSize = 0;
+    private String dbUrl = "/localhost:5432/javaBegin";
+    private String dbUser = "postgres";
+    private String dbPassword = "root";
+    private List<Connection> pool = new ArrayList<>(maxPoolSize);
+
+    public ConnectionPool() throws ClassNotFoundException, SQLException {
+        initPooledConnections();
+    }
 
-    private static BasicDataSource dataSource;
-
-    static ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
-    static InputStream input = classLoader.getResourceAsStream("config.properties");
-    static Properties prop = new Properties();
+    private void initPooledConnections() throws ClassNotFoundException, SQLException {
+        for (int i = 0; i < initialPoolSize; i++) {
+            openAndPoolConnection();
+        }
+    }
 
-    static {
+    private  void openAndPoolConnection() throws SQLException {
         try {
-            prop.load(input);
-        } catch (IOException e) {
+            Class.forName("org.postgresql.Driver");
```

Comment:
driver loading in the catch section - so, it may not loads. Make it as a part of initialization (in private constructor according to singleton pattern)